gst/edge/edge_sink.c

```diff
@@ -0,0 +1,367 @@
+ * @bug     No known bugs
+ *
+ */
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#include "edge_sink.h"
+
+GST_DEBUG_CATEGORY_STATIC (gst_edgesink_debug);
+#define GST_CAT_DEFAULT gst_edgesink_debug
+
+/**
+ * @brief the capabilities of the inputs.
+ */
+static GstStaticPadTemplate sinktemplate = GST_STATIC_PAD_TEMPLATE ("sink",
+    GST_PAD_SINK,
+    GST_PAD_ALWAYS,
+    GST_STATIC_CAPS_ANY);
+
+/**
+ * @brief edgesink properties
+ */
+enum
+{
+  PROP_0,
+
+  PROP_HOST,
+  PROP_PORT,
+  PROP_CONNECT_TYPE,
+
+  PROP_LAST
+};
+
+#define gst_edgesink_parent_class parent_class
+G_DEFINE_TYPE (GstEdgeSink, gst_edgesink, GST_TYPE_BASE_SINK);
+
+static void gst_edgesink_set_property (GObject * object,
+    guint prop_id, const GValue * value, GParamSpec * pspec);
+
+static void gst_edgesink_get_property (GObject * object,
+    guint prop_id, GValue * value, GParamSpec * pspec);
+
+static void gst_edgesink_finalize (GObject * object);
+
+static gboolean gst_edgesink_start (GstBaseSink * basesink);
+static GstFlowReturn gst_edgesink_render (GstBaseSink * basesink,
+    GstBuffer * buffer);
+static gboolean gst_edgesink_set_caps (GstBaseSink * basesink, GstCaps * caps);
+
+static gchar *gst_edgesink_get_host (GstEdgeSink * self);
+static void gst_edgesink_set_host (GstEdgeSink * self, const gchar * host);
+
+static guint16 gst_edgesink_get_port (GstEdgeSink * self);
+static void gst_edgesink_set_port (GstEdgeSink * self, const guint16 port);
+
+static nns_edge_connect_type_e gst_edgesink_get_connect_type (GstEdgeSink *
+    self);
+static void gst_edgesink_set_connect_type (GstEdgeSink * self,
+    const nns_edge_connect_type_e connect_type);
+
+/**
+ * @brief initialize the class
+ */
+static void
+gst_edgesink_class_init (GstEdgeSinkClass * klass)
+{
+  GObjectClass *gobject_class;
+  GstElementClass *gstelement_class;
+  GstBaseSinkClass *gstbasesink_class;
+
+  gstbasesink_class = (GstBaseSinkClass *) klass;
+  gstelement_class = (GstElementClass *) gstbasesink_class;
+  gobject_class = (GObjectClass *) gstelement_class;
+
+  gobject_class->set_property = gst_edgesink_set_property;
+  gobject_class->get_property = gst_edgesink_get_property;
+  gobject_class->finalize = gst_edgesink_finalize;
+
+  g_object_class_install_property (gobject_class, PROP_HOST,
+      g_param_spec_string ("host", "Host",
+          "A self host address to accept connection from edgesrc", DEFAULT_HOST,
+          G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
+  g_object_class_install_property (gobject_class, PROP_PORT,
+      g_param_spec_uint ("port", "Port",
+          "A self port address to accept connection from edgesrc.",
+          0, 65535, DEFAULT_PORT, G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
+  g_object_class_install_property (gobject_class, PROP_CONNECT_TYPE,
+      g_param_spec_enum ("connect-type", "Connect Type",
+          "The connections type between edgesink and edgesrc.",
+          GST_TYPE_EDGE_CONNECT_TYPE, DEFAULT_CONNECT_TYPE,
```

Comment:
Q: Are there any reasons why ```PROP_PORT``` uses a *local static* variable instead of a member variable of the ```gobject_class```?